capstone_project/src/main/java/com/google/sps/data/PostService.java

```diff
@@ -100,13 +102,13 @@ public void storePost(HttpServletRequest request) {
     Entity postEntity = new Entity("Post");
     String message = request.getParameter("text");
     String fileType = request.getParameter("file-type");
-    String fileBlobKey = uploadFile(request);
+    Optional<String> fileBlobKey = uploadFile(request);
 
     if (fileType == null || fileType.isEmpty()) {
       fileType = "none";
     }
 
-    postEntity.setProperty("fileBlobKey", fileBlobKey);
+    postEntity.setProperty("fileBlobKey", fileBlobKey.toString());
```

Comment:
I don't think this is quite right. This will store the toString representation of a Optional<String>. What you want to do is first check to see if fileBlobKey.isPresent(), if so then postEntity.setProperty("fileBlobKey", fileBlobKey.get());

You would also want to do something in the case that fileBlobKey.isPresent() if false. (In this case, you should handle it as if the string was null).

Having Optional helps with null checking and reminds the programmer that something can be null/missing and it should be checked prior to accessing whatever the Optional<> holds. This way we can avoid NPEs.

Reply:
Okay this makes a lot more sense now. Thank you!